src/main/kotlin/graphics/scenery/backends/vulkan/VulkanTexture.kt

```diff
@@ -362,121 +383,201 @@ open class VulkanTexture(val device: VulkanDevice,
 
         logger.debug("Updating {} with {} miplevels", this, mipLevels)
         if (mipLevels == 1) {
-            with(VU.newCommandBuffer(device, commandPools.Standard, autostart = true)) {
-                if(!initialised) {
-                    transitionLayout(stagingImage.image,
-                        VK_IMAGE_LAYOUT_PREINITIALIZED,
-                        VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL, mipLevels,
-                        srcStage = VK_PIPELINE_STAGE_HOST_BIT,
-                        dstStage = VK_PIPELINE_STAGE_TRANSFER_BIT,
-                        commandBuffer = this)
-                }
+            val block = !(gt?.usageType?.contains(Texture.UsageType.AsyncLoad) ?: false)
```

Comment:
This function should likely be refactored, it's gotten a bit out of hand.

Reply:
Happy to do that together @skalarproduktraum. That code is mostly yours and I don't know how best to refactor it